gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/ExploreService.java

```diff
@@ -103,14 +108,21 @@ public ExploreResponse explore(RequestContext requestContext, ExploreRequest req
     }
   }
 
-  private boolean isContextAnEntityType(ExploreRequest request) {
+  private boolean isContextAnEntityType(ExploreRequest request, RequestContext requestContext) {
+    if (entityTypesProvider
```

Comment:
This should wholly replace the prior impl which remains on lines 117-119. If the enum has a value that's not registered, it's not a real entity type and should not be treated as such.